ts/src/pro/gate.ts

```diff
@@ -361,25 +361,75 @@ export default class gate extends gateRest {
         //        }
         //    }
         //
+        this.helperForHandleTickerBidAsk ('bidask', client, message);
+    }
+
+    async helperForWatchTickersBidsAsks (symbols: Strings = undefined, methdoName: Str = undefined, params = {}): Promise<Tickers> {
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols);
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams (methdoName, undefined, params);
+        let subType = undefined;
+        [ subType, params ] = this.handleSubTypeAndParams (methdoName, undefined, params);
+        if (symbols === undefined) {
+            if (marketType === undefined) {
+                throw new ArgumentsRequired (this.id + ' ' + methdoName + ' requires symbols or set params["defaultType"] to any from "spot|swap|future". if you do not use spot, then you also need to set params["defaultSubType"] to "linear" or "inverse"');
+            }
+            this.checkRequiredArgument (methdoName, marketType, 'defaultType', [ 'spot', 'swap', 'future', 'option' ]);
+            const isSwapOrFuture = this.inArray (marketType, [ 'swap', 'future' ]);
+            if (isSwapOrFuture) {
+                this.checkRequiredArgument (methdoName, subType, 'defaultSubType', [ 'linear', 'inverse' ]);
+            }
+            let filteredMarkets = this.filterBy (this.markets, 'type', marketType);
+            filteredMarkets = this.filterByArray (filteredMarkets, 'active', [ true ], false);
+            if (isSwapOrFuture) {
+                filteredMarkets = this.filterBy (filteredMarkets, 'subType', subType);
+            }
+            symbols = this.getArrayOfObjectsKey (filteredMarkets, 'symbol');
+        }
+        const market = this.market (symbols[0]);
+        const messageType = this.getTypeByMarket (market);
+        const marketIds = this.marketIds (symbols);
+        let method = undefined;
+        [ method, params ] = this.handleOptionAndParams (params, methdoName, 'method');
+        const url = this.getUrlByMarket (market);
+        const channel = messageType + '.' + method;
+        const isWatchTickers = (methdoName === 'watchTickers');
+        const prefix = isWatchTickers ? 'ticker' : 'bidask';
+        const messageHashes = [];
+        for (let i = 0; i < symbols.length; i++) {
+            const symbol = symbols[i];
+            messageHashes.push (prefix + ':' + symbol);
+        }
+        const tickerOrBidAsk = await this.subscribePublicMultiple (url, messageHashes, marketIds, channel, params);
+        if (this.newUpdates) {
+            const items = {};
```

Comment:
I would have to test it in this context. But when looking to implement this on binance. I needed to create an ArrayCache, as it could happen that the future is resolved several times, but the user only receives it once. Therefore losing information. I'm worndering if we would have to do the same here.
https://github.com/ccxt/ccxt/pull/20903

Reply:
ok, lmk 👍

Reply:
@ttodua did you check if the scenario Pable describes happens here?

Reply:
it is global-wide problem, not specific to an exchange imo. if it needs to be fixed, it would be an isolated PR/fix, probably for whole lib, not specific to gate or binance. 

but i'll check anyway.

Reply:
i'm pushing last fix now, specific to this pr